src/classes/CW_AudienceCSVProcessController.cls

```diff
@@ -0,0 +1,131 @@
+/**
+* File Name   :    CW_AudienceCSVProcessController.cls
+* Description :    Page Contrloer for second step in importing Audience Rewards
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 20/05/2014 Alex Panka	  Intial 
+*/
+
+public with sharing class CW_AudienceCSVProcessController {
+	public final static Integer BATCH_SIZE=2000;
+	public CW_CSV_Import_Status__c impRevStat{get;private set;}
+	
+	public boolean IsProcessing{get;private set;}
+	public ID  batchJobId{get;private set;}
+	public integer countTiks{get;set;}
+	public AsyncApexJob batchJob {get; private set;}
+	private ID CsvStatRecId;
+	
+	public CW_AudienceCSVProcessController(){
+		CsvStatRecId=(ID) Apexpages.currentPage().getParameters().get('ID');
+		if(CsvStatRecId!=null){
+			impRevStat= [Select Schedule__c, Reward_Catalogue__c, NoRecords__c, NoRecToUpload__c, Name, MissingGroup__c, MissingFreeBet__c, MissingContact__c, Marketing_Campaign__c, Id From CW_CSV_Import_Status__c WHERE ID=:CsvStatRecId ];
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.